sources/niUI/src/VGPaint.cpp

```diff
@@ -209,7 +209,8 @@ class cVGGradientTable : public cIUnknownImpl<iVGGradientTable>
       trans.invert();
     }
     tAGGLinearInterpolator  inter(trans);
-    tAGGGradientSpanGen   span_gen(inter, *ptrGrad, AGGGetGradientTable(this), agg_real(anD1), agg_real(anD2));
+    sAGGGradientTable t = AGGGetGradientTable(this);
+    tAGGGradientSpanGen  span_gen(inter, *ptrGrad, t, agg_real(anD1), agg_real(anD2));
```

Comment:
Is that change still necessary? I'm assuming `t` was added to make debugging easier?

Reply:
Yes, the code will crash with out it, cause the gradient table gets released before we render call

Reply:
oh... does it take a reference to the object? tAGGGradientSpanGen's constructor should be updated to ask for a moved object probably (&&) so that the compiler complains when using a temporary value.

The AGG code is ancient C++ at this point xD

Reply:
```
  span_gradient(interpolator_type& inter,
                const GradientF& gradient_function,
                const ColorF& color_function,
                agg_real d1, agg_real d2) :
      m_interpolator(&inter),
      m_gradient_function(&gradient_function),
      m_color_function(&color_function),
      m_d1(iround(d1 * gradient_subpixel_scale)),
      m_d2(iround(d2 * gradient_subpixel_scale))
  {}
...
  interpolator_type* m_interpolator;
  const GradientF*   m_gradient_function;
  const ColorF*      m_color_function;
  int                m_d1;
  int                m_d2;
```
...beautiful take a pointer of a const ref parameter...

Ideally all that should be astl::non_null<> pointers so its clear what happens, but that lib is from 2005 so we can forgive it not using modern C++...